src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.GeneratedFileReplacingCompilationTracker.cs

```diff
@@ -162,7 +162,7 @@ private async Task<Checksum> ComputeDependentChecksumAsync(SolutionCompilationSt
                 {
                     // The generated file still exists in the underlying compilation, but the contents may not match the open file if the open file
                     // is stale. Replace the syntax tree so we have a tree that matches the text.
-                    newStates = newStates.SetState(id, replacementState);
+                    newStates = newStates.SetState(replacementState);
```

Comment:
passing 'id' is redundant.  it's already retrievable off of hte stat eobject passed in.